src/main/java/org/tron/common/runtime/RuntimeImpl.java

```diff
@@ -141,29 +128,29 @@ public RuntimeImpl(Transaction tx, BlockCapsule block, DepositImpl deposit,
     this.isStaticCall = isStaticCall;
   }
 
-  public RuntimeImpl(Transaction tx, BlockCapsule block, DepositImpl deposit,
+  private RuntimeImpl(Transaction tx, BlockCapsule block, DepositImpl deposit,
       ProgramInvokeFactory programInvokeFactory) {
     this.trx = tx;
     this.deposit = deposit;
     this.programInvokeFactory = programInvokeFactory;
-    this.executorType = ET_PRE_TYPE;
+    this.executorType = ExecutorType.ET_PRE_TYPE;
     this.blockCap = block;
     this.energyProcessor = new EnergyProcessor(deposit.getDbManager());
     ContractType contractType = tx.getRawData().getContract(0).getType();
     switch (contractType.getNumber()) {
       case ContractType.TriggerSmartContract_VALUE:
-        trxType = TRX_CONTRACT_CALL_TYPE;
+        trxType = TrxType.TRX_CONTRACT_CALL_TYPE;
         break;
       case ContractType.CreateSmartContract_VALUE:
-        trxType = TRX_CONTRACT_CREATION_TYPE;
+        trxType = TrxType.TRX_CONTRACT_CREATION_TYPE;
         break;
       default:
-        trxType = TRX_PRECOMPILED_TYPE;
+        trxType = TrxType.TRX_PRECOMPILED_TYPE;
     }
   }
 
 
-  public void precompiled() throws ContractValidateException, ContractExeException {
+  private void precompiled() throws ContractValidateException, ContractExeException {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the redundant '!unknownSymbol!' thrown exception declaration(s). ([squid:RedundantThrowsDeclarationCheck](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3ARedundantThrowsDeclarationCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1681&issues=AWbnEHMySgN4uCcljsrs&open=AWbnEHMySgN4uCcljsrs)

[ISSUE_KEY]: AWbnEHMySgN4uCcljsrs